### PR TITLE
Shared | Autogen: Adding support for standalone components

### PR DESCRIPTION
**Pull request by reb-dev**
This PR updates component autogeneration logic so that all Unovis components (not containers) can be automatically generated when running `npm run generate`. Previously, our standalone components BulletLegend, LeafletMap and LeafletFlowMap had to be updated manually in the React and Angular wrappers.

In addition to autogen, another motivation for this change is to ensure our list in `shared/integrations/components` is complete. This allows the list be used for multiple purposes across the codebase (docs, commitlint, etc).

closes: unovis/issues-only#41

### CODE REVIEW

File: packages/react/autogen/component.ts [hunk 1]
```diff
@@ -5,61 +5,66 @@ export function getComponentCode (
...
-    const component = componentRef.current
+    ${isStandAlone ? '' : 'const component = componentRef.current'}
     ${dataType ? 'if (props.data) component?.setData(props.data)' : ''}
-    component?.setConfig(props)
+    component?.${componentName === 'BulletLegend' ? 'update ' : 'setConfig'}(props)
   })
 
-  useImperativeHandle(fRef, () => ({ get component () { return componentRef.current } }), [])
-  return <vis-${elementSuffix} ref={ref} />
+  useImperativeHandle(fRef, () => ({ get component () { return component${isStandAlone ? '' : 'Ref.current'} } }), [])
```
Comment (rokotyan) [inline packages/react/autogen/component.ts:66]: The `Ref.current`statement looks a bit suspicious, is it correct?

**Review by rokotyan — APPROVED** [review]
Thanks for implementing this @reb-dev!

I'm also planning to allow stand-alone components like Bullet Legend to be appended straight to the provided DOM node to simplify styling when used with UI frameworks like React. Currently we have double divs — one comes from the framework (e.g. via a ref), and another one is created by a component.

For example, here's a Bullet Legend change that I've already tested with a published beta:
```ts
constructor (element: HTMLElement, config?: BulletLegendConfigInterface) {
    this._container = element
    this.div = config?.renderIntoProvidedDomNode ? select(this._container) : select(this._container).append<HTMLElement>('div')
...
```

And the the React wrapper the initialization looks like this:
```ts
const c = new BulletLegend(container.current as HTMLDivElement, { ...props, renderIntoProvidedDomNode: true })
```

If I create a Pull Request with these changes, will you be able to incorporate them in this autogen update too?

File: packages/react/autogen/component.ts (hunk 1 above)
Comment (reb-dev) [reply packages/react/autogen/component.ts]: Nah that is definitely a typo, thanks for catching it. Updated.

**Comment by reb-dev** [comment]
> If I create a Pull Request with these changes, will you be able to incorporate them in this autogen update too?

@rokotyan Sure 👍

**Review by lee00678 — APPROVED** [review]

**Pull request merged**